temporal-sdk/src/main/java/io/temporal/client/WorkflowStartOperationUpdate.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.client;
+
+import io.temporal.workflow.Functions;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.concurrent.*;
+
+/** TBD */
+public class WorkflowStartOperationUpdate<R>
```

Comment:
IMO it doesn't make sense to have `Workflow` + `Start` + `Operation` + `Update` extend `Start` + `Workflow` + `Additional` + `Operation` with the adjectives inconsistent or missing.

Like `WorkflowOptions` are not `WorkflowStartOptions` because start is implied, `StartWorkflowAdditionalOperation` may be able to imply `Start`. Some possible options:

* `UpdateWorkflowAdditionalOperation` extends `WorkflowAdditionalOperation`
* `AdditionalUpdateWorkflowOperation` extends `AdditionalWorkflowOperation`
* `UpdateWithStartWorkflowAdditionalOperation` extends `StartWorkflowAdditionalOperation`
* `AdditionalUpdateWithStartWorkflowOperation` extends `AdditionalStartWorkflowOperation`

Or if we have decided to remove the term "Additional", then we should remove it from the base class too (though not as much of a fan, we should decide whether the adjective is generally valuable or not, not selectively), e.g. 

* `StartWorkflowOperationUpdate` extends `StartWorkflowOperation`
* `WorkflowUpdateStartOperation` extends `WorkflowStartOperation`

Or something like that. No strong opinion here.